src/std/db/postgresql-driver.ss

```diff
@@ -331,31 +378,98 @@
        (resync!)
        (apply raise-sql-error 'postgresql-query! msg irritants))))
 
+  ;; Query backpressure mechanism: the query pump reads and buffers up
+  ;; to query-limit rows before requiring a continue signal from
+  ;; the query client.
+  ;; If the client aborts (or issues a new query), the remaining results
+  ;; of the query will be skipped over on resync.
+  ;; The plan was originally to use (named) portals with Execute limit
+  ;; and rely on PortalSuspended to implement staging of input.
+  ;; Unfortunately, the server (tested with 10.1) doesn't start
+  ;; sending anything back until it sees a Sync; this kills
+  ;; the portal unless it's in a BEGIN/COMMIT block so the backpressure
+  ;; idea using portal suspension can't work without wrapping every
+  ;; query in an explicit txn (which is undesirable for obvious reasons)
+
   (def (query-pump)
-    ;; Execute ("")          -> DataRow ...
-    ;;                          CommandComplete | EmptyQueryResponse
-    ;;                          | ErrorResponse | PortalSuspended.
-    ;; Sync                  -> ReadyForQuery
+      ;; Execute ("")          -> DataRow ...
+      ;;                          CommandComplete | EmptyQueryResponse
+      ;;                          | ErrorResponse | PortalSuspended.
+      ;; Sync                  -> ReadyForQuery
+      (let/cc break
+        (let lp ()
+          (match (recv!)
+            (['DataRow . cols]
+             (cond
+              ((channel-try-put query-output cols)
+               (lp))
+              (else
+               (channel-sync query-output cols query-token)
+               (break))))
+            (['CommandComplete tag]
+             (channel-sync query-output (void))
```

Comment:
why do we need the void sync here? can you add a comment?

Reply:
Good eye. At one point the statement {query-fetch} itself did not yet know about CommandComplete at it was all set in the postgresql-query {query-fetch} so the void was akin to EOF. 

I think it can be removed and save a byte. I'll check on that today.

Reply:
Force pushed.